client/components/data/query-posts/README.md

```diff
@@ -0,0 +1,53 @@
+Query Posts
+===========
+
+Query Posts is a React component used in managing the fetching of posts queries.
+
+## Usage
+
+Render the component, passing `siteId` and `query`. It does not accept any children, nor does it render any elements to the page. You can use it adjacent to other sibling components which make use of the fetched data made available through the global application state.
```

Comment:
>Can we make `siteId` determine whether to fetch single site or all sites?

I think that's a reasonable enhancement, though also requires additional action creators in the sites state to handle fetching posts from `/me` endpoints. Seems appropriate for a subsequent pull request.